pgmuvi/lightcurve.py

```diff
@@ -393,30 +395,64 @@ def print_results(self):
                     print(f"{key}: {results_tmp.flatten()}")
                 elif sum_over_shape == 2:
                     print(f"{key}: {results_tmp[:,0].flatten()}, {results_tmp[:,1].flatten()}")
-            
+           
 
-    def plot_psd(self, means, freq, scales, weights, show=True):
-        #Computing the psd for frequencies f
-        psd = compute_psd(means, freq, scales, weights)
+    def plot_psd(self, means, freq, scales, weights, show=True, wavelengths=None):
+        if means.ndim == 1:  # 1D case
+            # Computing the psd for frequencies freq
+            psd = compute_psd_1D(means, freq, scales, weights)
 
-        # Initialize plot
-        fig, ax = plt.subplots(1, 1, figsize=(8, 6))
+            # Initialise plot
+            fig, ax = plt.subplots(1, 1, figsize=(8, 6))
 
-        #plotting psd
-        ax.plot(freq, psd)
-        if show:
-            plt.show()
-        else:
-            return fig, ax
-
-    def compute_psd(means, freq, scales, weights):
-        c = np.zeros((len(means),) + freq.shape, ) #mean = mean of each gaussian in the psd (the kernel we use uses only gaussians).
-        for i, m in enumerate(means): #f = array of frequencies that we want to plot
-            s = scales[i] #s.d
-            w = weights[i] #how much power is given to each gaussian
-            c[i] = np.sqrt(w) * (norm.pdf(freq, m, s) - norm.pdf(-freq, m, s))  #subtracting negative side of psd - otherwise it would cause interference
-            # Each component of the PSD is the weight times the difference of the forward and reverse PDFs
-            # In this case, the weights are square-rooted, because the original AGW formula for the kernel uses weights**2 while gpytorch implements weights, and therefore we must adjust our interpretation of the output.
+            # plotting psd
+            ax.plot(freq, psd)
+            if show:
+                plt.show()
+            else:
+                return fig, ax
+
+    if means.ndim == 2:  # 2D case
```

Comment:
I'm not sure if the indentation here is correct - it looks like it might have once level of indentation too few, but I'm getting confused by all my previous comments...